apps/api/src/app/shared/shared.module.ts

```diff
@@ -109,7 +110,7 @@ const PROVIDERS = [
   {
     provide: ANALYTICS_SERVICE,
     useFactory: async () => {
-      const analyticsService = new AnalyticsService();
+      const analyticsService = new AnalyticsService(process.env.SEGMENT_TOKEN);
```

Comment:
moved the segment token to be passed to the constructor, before the logic was hardcoded in the service itself